<launch>
    <arg name="laser_topic" default="/scan"/>
    <arg name="wheelchair_detection_topic" default="/wheelchair_detections"/>
    <arg name="walker_detection_topic" default="/walker_detections"/>
    <arg name="class_agnostic_detection_topic" default="/mobility_aid_detections"/>
    <arg name="detection_threshold" default="0.9"/>
    <arg name="use_cudnn" default="false" />
    <arg name="network_param_file" default="$(find wheelchair_detector)/resources/net-wcnet.npz"/>

    <arg name="machine" default="localhost"/>
    <arg name="user" default=""/>

    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>
    
    <node pkg="wheelchair_detector" type="detector.py" name="detector" output="log" respawn="true">
        <param name="laser_topic" value="$(arg laser_topic)" type="string"/>
        <param name="wheelchair_detection_topic" value="$(arg wheelchair_detection_topic)" type="string"/>
        <param name="walker_detection_topic" value="$(arg walker_detection_topic)" type="string"/>
        <param name="class_agnostic_detection_topic" value="$(arg class_agnostic_detection_topic)" type="string"/>
        <param name="detection_threshold" value="$(arg detection_threshold)" type="double"/>
        <param name="use_cudnn" value="$(arg use_cudnn)" type="bool"/>
        <param name="network_param_file" value="$(arg network_param_file)" type="string"/>
    </node>

</launch> 